### PR TITLE
Creating new user state client interface

### PR DESCRIPTION
**Pull request by utkjad**
This PR adds
django ORM (MySQL) to  /edx-user-state-client 

The current structure is -
```
├── LICENSE
├── README.md
├── __init__.py
├── backends ---> A directory which handles all backends
│   └── django ---> handles djangoORM (MySQL)
│       ├── __init__.py
│       
└── interface.py
```
Have made changes accordingly in `edx/edx-platform` repo.
Link to PR-  https://github.com/edx/edx-platform/pull/8650

### CODE REVIEW

**Comment by utkjad** [comment]
Will add this repo as an edx platform dependency after the merge.
Link to the pull req-

**Comment by utkjad** [comment]
@cpennington  @doctoryes  @alawibaba

File: backends/djangoORM-user-state-client/user_state_client.py [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+from xblock.fields import Scope, ScopeBase
+from xblock_user_state.interface import XBlockUserStateClient
+from courseware.models import StudentModule, StudentModuleHistory
+from contracts import contract, new_contract
+from opaque_keys.edx.keys import UsageKey
+
+new_contract('UsageKey', UsageKey)
+
+
+class DjangoXBlockUserStateClient(XBlockUserStateClient):
```
Comment (cpennington) [inline backends/djangoORM-user-state-client/user_state_client.py]: Let's not move this one yet.

**Comment by cpennington** [comment]
`djangoORM-user-state-client` isn't a good name for a python module. The name of the directory maps to what you import, and this won't work in python `import backends.djangoORM-user-state-client`. Instead, when we move the django backend, I would just use a module named `backends.django`.

**Comment by utkjad** [comment]
Renamed to `django`

File: backends/djangoORM-user-state-client/user_state_client.py (hunk 1 above)
Comment (utkjad) [inline backends/djangoORM-user-state-client/user_state_client.py]: removed from `edx/edx-user-state-client`, and added to `edx/edx-platform`

**Pull request merged**